articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,547 @@
+We have discussed how to use the default Winston logger to generate app logs. Winston allows you to create [custom loggers](https://github.com/winstonjs/winston#adding-custom-transports). This gives you the power to control and customize specific loggers based on your application services.
+
+For example:
+
+```js
+const { createLogger, format, transports, config } = require('winston');
+
+const usersLogger = createLogger({
+
+   levels: config.syslog.levels,
+   format: combine(
+       timestamp({
+            format: 'YYYY-MM-DD HH:mm:ss'
+}),
+
+   transports: [
+        new transports.File({ filename: 'users.log' })
+     ]
+});
+const transactionLogger = createLogger({
+   transports: [
+        new transports.File({ filename: 'transaction.log' })
+     ]
+});
+
+module.exports = {
+    usersLogger: usersLogger,
+    transactionLogger: transactionLogger
+};
+```
+
+In the above example, the application has two services, `users` and `transactions`. Creating different loggers for such services will be relevant. These distinguish an issue for various application services.
+
+For example, assuming a `user.js` file that processes user services' properties, the following example will apply for a simple log.
+
+```js
+// Require logger.js
+const {usersLogger, transactionLogger} = require('./logger');
+
+...........// Your users' code setups.................
+
+// What to record to users.log
+usersLogger.info('New user created');
+usersLogger.error(`Unable to find user: ${err}`);
+```
+
+In an actual application, we can create the following logs when a new user is created or when a new transaction takes place in the system. The attributes will be extracted and recorded in the log file.
+
+user.js
+
+```js
+// Require logger.js
+const {usersLogger, transactionLogger} = require('./logger');
+...........// Your users' code setups.................
+usersLogger.info('User created!',
+     {user_id: `${res._id}`,
+     user_name: `${res.name}`,
+     user_email: `${res.email}`,
+);
+usersLogger.error(`Unable to find user: ${err}`);
+```
+
+transaction.js
+
+```js
+// Require logger.js
+const {usersLogger, transactionLogger} = require('./logger');
+
+...........// Your Transaction' code setups.................
+
+// Transaction logs
+transactionLogger.info('Session connected', { session_id: `${req.id}`}, {user_id: `${res._id}`);
+transactionLogger.info('Transaction Initiated', { transaction_id: `${req.transaction_id}`}, {user_id: `${res.user_id}`);
+transactionLogger.info('Transaction completed',  {user_id: `${res.user_id}`, {`${req.transaction_amout}`, `${req.transaction_code}`});
+transactionLogger.error('Transaction Failed', `{${err}`,session_id: `${req.id}`}, {user_id: `${res._id}`);
+```
+
+You will have easier logs management. Thus easier to debug an issue for a specific user account and transactions using user attributes such as `name`, `email`, and `id`. Transaction attributes as `transaction_id` and `transaction_code`.
```

Comment:
**Flow Edit:** This makes it easier to manage your logs because there are seperated for specific issues and user and transaction attributes are provided to help narrow it down to a specific user or transaction.